frontend/include/chpl/uast/When.h

```diff
@@ -33,42 +34,40 @@ namespace uast {
   This class represents a when statement. When statements make up the body
   of the select statement.
  */
-class When final : public SimpleBlockLike {
+class When final : public AstNode {
  friend class AstNode;
 
  private:
   // The position of this never changes.
   static const int8_t caseExprChildNum_ = 0;
-
-  int numCaseExprs_;
-
-  When(AstList children, int numCaseExprs, BlockStyle blockStyle,
-       int bodyChildNum,
-       int numBodyStmts)
-    : SimpleBlockLike(asttags::When, std::move(children), blockStyle,
-                      bodyChildNum,
-                      numBodyStmts),
-      numCaseExprs_(numCaseExprs) {
+  const int numCaseExprs_;
+  const BlockStyle blockStyle_;
+  
+  When(AstList children, int numCaseExprs, BlockStyle blockStyle)
+    : AstNode(asttags::When, std::move(children)),
+      numCaseExprs_(numCaseExprs),
+      blockStyle_(blockStyle) {
   }
 
   void serializeInner(Serializer& ser) const override {
-    simpleBlockLikeSerializeInner(ser);
+    serializeChildren(ser);
```

Comment:
I don't think this should be necessary. Does removing this line cause any tests to fail?

Reply:
It does not; i'll remove it!